### PR TITLE
Alberto

### PR DESCRIPTION
**Pull request by afigallo**
UserController - Patch

### CODE REVIEW

**Comment by ryanmohler172** [comment]
Can one of the admins verify this patch? 
Run `jenkins approve` to approve this pr for building. Run `jenkins build` to build.
You can also run `jenkins whitelist` to whitelist this user for all future prs.

**Comment by ryanmohler17** [comment]
jenkins whitelist

**Review by ARosenbaum98 — APPROVED** [review]

**Comment by ryanmohler172** [comment]
[63% (-13.64%) vs master 76%](http://3.91.248.52:8080/jenkins/job/User-Service%20-%20PR/41/)

**Comment by ryanmohler172** [comment]
[67% (-9.57%) vs master 76%](http://3.91.248.52:8080/jenkins/job/User-Service%20-%20PR/42/)

File: src/main/java/com/reverse/userservice/controllers/UserController.java [hunk 1]
```diff
@@ -100,4 +103,36 @@ public ResponseEntity editUser(@RequestBody UserEdit userEdit) {
...
+                switch (field){
+                    case "username": if (user.getUsername() != null) updUser.setUsername(user.getUsername()); break;
+                    case "email": if (user.getEmail() != null) updUser.setEmail(user.getEmail()); break;
+                    case "firstName": if (user.getFirstName() != null) updUser.setFirstName(user.getFirstName()); break;
+                    case "lastName": if (user.getLastName() != null) updUser.setLastName(user.getLastName()); break;
+                    case "password": if (user.getPassword() != null) updUser.setPassword(user.getPassword()); break;
+                    case "dateOfBirth": if (user.getDateOfBirth() != null) updUser.setDateOfBirth(user.getDateOfBirth()); break;
+                    case "gender": if (user.getGender() != null) updUser.setGender(user.getGender()); break;
+                    case "branch": if (user.getBranch() != null) user.setBranch(user.getBranch()); break;
+                    case "profilePicture": if (user.getProfilePicture() != null) user.setProfilePicture(user.getProfilePicture()); break;
```
Comment (CJacobJ) [inline src/main/java/com/reverse/userservice/controllers/UserController.java]: I've got fixed tests added, but it looks like there is a mistake in the code here. I think these are supposed to be setting the Branch and profilePicture of the updUser.
```suggestion
                    case "branch": if (user.getBranch() != null) updUser.setBranch(user.getBranch()); break;
                    case "profilePicture": if (user.getProfilePicture() != null) updUser.setProfilePicture(user.getProfilePicture()); break;
```

**Review by afigallo — COMMENTED** [review]
Reviewed. Test passed and has more lines covered. **Thanks Jacob!**

**Comment by ryanmohler172** [comment]
[80% (+3.43%) vs master 76%](http://3.91.248.52:8080/jenkins/job/User-Service%20-%20PR/45/)

**Pull request merged**